src/system/documents/item.ts

```diff
@@ -1051,18 +1049,83 @@ export class CosmereItem<
 
     /* --- Helpers --- */
 
-    protected async getEnrichedDescription(): Promise<string | undefined> {
-        if (!this.hasDescription()) return;
-        if (
-            !(this as CosmereItem<DescriptionItemData>).system.description
-                ?.value
-        )
-            return;
-
-        return await TextEditor.enrichHTML(
-            (this as CosmereItem<DescriptionItemData>).system.description!
-                .value!,
+    protected async getDescriptionHTML(): Promise<string | undefined> {
+        if (!this.hasDescription()) return undefined;
+
+        const descriptionData = (this as CosmereItem<DescriptionItemData>)
+            .system.description?.value;
+        const description = await TextEditor.enrichHTML(descriptionData ?? '');
+
+        const traitsNormal = [];
+        const traitsExpert = [];
+        const traits = [];
+        if (this.system.traits) {
+            for (const [key, value] of Object.entries(
+                this.system.traits as TraitsItemData,
```

Comment:
The cast here is incorrect. `TraitsItemData` is the entire data model type, i.e. the system is of type `TraitsItemData`, which has a property called `traits`. If you use the above type guard this entire cast can be removed. That should also allow you to drop the eslint-disables in the next bit.

Reply:
Thanks for the headsup, still new to Typescript so sometimes it breaks my brain trying to fix the lint errors 😂